From 5cfe5fb175aa7c4c837b57b5a99074157cd5958b Mon Sep 17 00:00:00 2001
From: Hendrik <git@h3ndrk.de>
Date: Thu, 28 Apr 2022 15:10:53 +0200
Subject: [PATCH] Rename thread_local target option and disable it

Since the symbol `__FastLocalKeyInner` is not existant, we need to disable this
option.

Commit that renames the target options:
https://github.com/rust-lang/rust/commit/391332c5d9d5a5e97a0d36e011a87ad43045cfd3

One relevant section where the non existant symbol is used:
https://github.com/rust-lang/rust/blob/3bfeffd55bbc7d653d2df7ad265746bafe595a96/library/std/src/thread/local.rs#L323-L329
---
 meta/recipes-devtools/rust/rust-common.inc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meta/recipes-devtools/rust/rust-common.inc b/meta/recipes-devtools/rust/rust-common.inc
index 310aecef22..c9877b1c11 100644
--- a/meta/recipes-devtools/rust/rust-common.inc
+++ b/meta/recipes-devtools/rust/rust-common.inc
@@ -352,7 +352,7 @@ def rust_gen_target(d, thing, wd, features, cpu, arch, abi=""):
     tspec['linker-is-gnu'] = True
     tspec['linker-flavor'] = "gcc"
     tspec['has-rpath'] = True
-    tspec['has-elf-tls'] = True
+    tspec['has-thread-local'] = False
     tspec['position-independent-executables'] = True
     tspec['panic-strategy'] = d.getVar("RUST_PANIC_STRATEGY")
 
-- 
2.30.2

